R/load_plotinfo.R

```diff
@@ -1,45 +1,53 @@
-#'
-#' @param database name of fieldmap/access database (with specific fieldmap
-#' structure) including path
-#'
-#' @return Dataframe with columns plot_id, plottype and forest_reserve
-#'
-#' @examples
-#' \dontrun{
-#' #change path before running
-#' library(forrescalc)
-#' load_plotinfo("C:/MDB_BOSRES_selectieEls/FieldMapData_MDB_BOSRES_selectieEls.accdb")
-#' }
-#'
-#' @export
-#'
-#' @importFrom RODBC odbcClose odbcConnectAccess2007 sqlQuery
-#' @importFrom dplyr %>% bind_rows distinct
-#'
-load_plotinfo <- function(database) {
-  query_plot <-
-    "SELECT Plots.ID AS plot_id,
-      Plots.Plottype AS plottype,
-      pd.ForestReserve AS forest_reserve
-    FROM Plots INNER JOIN PlotDetails_1eSet pd ON Plots.ID = pd.IDPlots;"
-
-  query_plot2 <-
-    "SELECT Plots.ID AS plot_id,
-      Plots.Plottype AS plottype,
-      pd.ForestReserve AS forest_reserve
-    FROM Plots INNER JOIN PlotDetails_2eSet pd ON Plots.ID = pd.IDPlots;"
-
-  con <- odbcConnectAccess2007(database)
-  plotinfo <- sqlQuery(con, query_plot, stringsAsFactors = FALSE) %>%
-    bind_rows(
-      sqlQuery(con, query_plot2, stringsAsFactors = FALSE)
-    ) %>%
-    distinct()
-  odbcClose(con)
-
-  return(plotinfo)
-}
+#' retrieve generic plot data from fieldmap database
+#'
+#' This function queries the given database to retrieve additional data on plots
+#' to save in forresdat and link with the datasets that are saved there.
+#'
+#' @param database name of fieldmap/access database (with specific fieldmap
+#' structure) including path
+#'
+#' @return Dataframe with columns plot_id, plottype and forest_reserve
+#'
+#' @examples
+#' \dontrun{
+#' #change path before running
+#' library(forrescalc)
+#' load_plotinfo("C:/MDB_BOSRES_selectieEls/FieldMapData_MDB_BOSRES_selectieEls.accdb")
+#' }
+#'
+#' @export
+#'
+#' @importFrom dplyr %>% distinct filter group_by mutate left_join select summarise ungroup
+#' @importFrom rlang .data
+#'
+
+load_plotinfo <- function(database) {
+  query_plot <-
+    "SELECT Plots.ID AS plot_id,
+      Plots.Plottype AS plottype,
+      pd.ForestReserve AS forest_reserve,
+      pd.Survey_Trees_YN AS survey_trees,
+      pd.Survey_Deadwood_YN AS survey_deadw,
+      pd.Survey_Vegetation_YN AS survey_veg,
+      pd.Survey_Regeneration_YN AS survey_reg,
+      pd.DataProcessed_YN AS data_processed
+    FROM Plots INNER JOIN PlotDetails_%1$deSet pd ON Plots.ID = pd.IDPlots;"
+
+  plotinfo <-
+    query_database(database, query_plot) %>%
+    distinct()
+
+  plotinfo2 <- plotinfo %>%
+    left_join(plotinfo %>%
+                filter(.data$survey_trees == 10) %>%
+                group_by(.data$plot_id, .data$plottype, .data$forest_reserve, .data$survey_trees) %>%
+                summarise(min_period = min(.data$period)) %>%
+
+                ungroup()) %>%
+    mutate(survey_number = .data$period - .data$min_period + 1) %>%
+    select(-.data$min_period)
+
+  plotinfo <- plotinfo2

```

Comment:
```suggestion
  plotinfo <- plotinfo %>%
    left_join(plotinfo %>%
                filter(.data$survey_trees == 10) %>%
                group_by(.data$plot_id, .data$plottype, .data$forest_reserve, .data$survey_trees) %>%
                summarise(min_period = min(.data$period)) %>%

                ungroup()) %>%
    mutate(survey_number = .data$period - .data$min_period + 1) %>%
    select(-.data$min_period)
```

Ach ja, nog een kleine vereenvoudiging die ik in mijn suggestie-branch vergeten was: op die manier moet je geen tweede variabele gebruiken (= extra geheugenruimte die je inneemt).

Reply:
OK, was ter controle dat ik dat zo nog had staan! (als join dan toch niet OK bleek, moest ik ook plotinfo opnieuw aanmaken..., maar beter dat dat in finale versie opgekuist is)

Reply:
Inderdaad, volledig akkoord: praktisch om dit in de testfase zo te doen een achteraf op te kuisen.  (Ik weet niet of je de commit suggestion al uitgetest had?  Dan is het nu je kans ;-) )